src/Security/MemberAuthenticator/LoginForm.php

```diff
@@ -82,7 +87,9 @@ public function __construct(
             $backURL = Session::get('BackURL');
         }
 
-        if ($checkCurrentUser && Member::currentUser() && Member::logged_in_session_exists()) {
+        if ($checkCurrentUser && Security::getCurrentUser() && Member::logged_in_session_exists()) {
```

Comment:
I think Member::logged_in_session_exists() is redundant now; Security::getCurrentUser() will never return a user that doesn't exist in the DB.